all_test.go

```diff
@@ -259,16 +273,19 @@ func TestResetPassword(t *testing.T) {
 	}
 
 	// Test time expiry
-	token3, _ := c.ResetPasswordStart("joe", time.Now().Add(-3*time.Second))
-	if err := c.ResetPasswordFinish("joe", token3, "12345"); err != ErrPasswordTokenExpired {
+	token3, _ := c.ResetPasswordStart(joeUserId, time.Now().Add(-3*time.Second))
+	if err := c.ResetPasswordFinish(joeUserId, token3, "12345"); err != ErrPasswordTokenExpired {
 		t.Fatalf("ResetPasswordFinish should have failed with ErrPasswordTokenExpired instead of %v", err)
 	}
+
+	teardown(c)
 }
 
 func TestBasicAuth(t *testing.T) {
 	c := setup1(t)
 
-	expect_username_password := func(username, password, expectErrorStart, expectIdentity string) {
+	expect_username_password := func(username, password, expectErrorStart string, expectIdentity string) {
```

Comment:
No need to make this change to the function definition. The original version was correct, and idiomatic.